tests/test_operator.py

```diff
@@ -0,0 +1,1179 @@
+    driver._send.assert_called_once_with({
+        'type': 'WAIT_NESTED_CONTAINER',
+        'wait_nested_container': {
+            'container_id': {
+                'parent': {
+                    'value': parent_id,
+                },
+                'value': container_id,
+            },
+        },
+    })
+
+
+def test_kill_nested_container(mocker):
+    agent = mocker.Mock()
+    driver = MesosOperatorAgentDriver(agent)
+    driver._send = mocker.Mock()
+    container_id = str(uuid.uuid4())
+    parent_id = str(uuid.uuid4())
+    driver.killNestedContainer(container_id, parent_id)
+    driver._send.assert_called_once_with({
+        'type': 'KILL_NESTED_CONTAINER',
+        'kill_nested_container': {
+            'container_id': {
+                'parent': {
+                    'value': parent_id,
+                },
+                'value': container_id,
+            },
+        },
+    })
+
+
+def test_launch_nested_container_session(mocker):
+    agent = mocker.Mock()
+    driver = MesosOperatorAgentDriver(agent)
+    driver._send = mocker.Mock()
+    launch_nested_container_session = {
+        'container_id': {
+            'parent': {
+                'parent': {
+                    'value': str(uuid.uuid4())
+                },
+                'value': str(uuid.uuid4())
+            },
+            'value': str(uuid.uuid4())
+        },
+        'command': {
+            'environment': {
+                'variables': [
+                    {
+                        'name': 'ENV_VAR_KEY',
+                        'type': 'VALUE',
+                        'value': 'env_var_value'
+                    }
+                ]
+            },
+            'shell': 'true',
+            'value': 'while [ true ]; do echo $(date); sleep 1; done'
+        }
+    }
+    driver.launchNestedContainerSession(launch_nested_container_session)
+    driver._send.assert_called_once_with({
+        'type': 'LAUNCH_NESTED_CONTAINER_SESSION',
+        'launch_nested_container_session': launch_nested_container_session,
+    }, headers={'Accept': 'application/recordio'})
+
+
+def test_attach_container_input(mocker):
+    agent = mocker.Mock()
+    driver = MesosOperatorAgentDriver(agent)
+    driver._send = mocker.Mock()
+    container_id = 'da737efb-a9d4-4622-84ef-f55eb07b861a'
+    process_ios = [
+        {
+            'type': 'DATA',
+            'data': {
+                'type': 'STDIN',
+                'data': 'dGVzdAo='
+            }
+        }
+        ,
+        {
+            'type': 'CONTROL',
+            'control': {
+                'type': 'HEARTBEAT',
+                'heartbeat': {
+                    'interval': {
+                        'nanoseconds': 30000000000
+                    }
+                }
+            }
+        }
+    ]
+    driver.attachContainerInput(container_id, process_ios)
+    awaited_result = '153\n{"type": "ATTACH_CONTAINER_INPUT", "attach_container_input": {"type": "CONTAINER_ID", ' + \
```

Comment:
Tests failed here: the order of `type` field and `container_id` field is indeterministic.
We may do some explicit checking here?

Reply:
I corrected the test case to get the same ordering in the output json.